others/Coppersmith_frievald's.cpp

```diff
@@ -0,0 +1,163 @@
+/**
+ * @file
+ * @brief Implementation of [Freivalds' algorithm]
+ * (https://en.wikipedia.org/wiki/Freivalds%27_algorithm)
+ * @author [Darshana Sarma](https://github.com/Darshana-Sarma)
+ * @details
+ * Given three n Ã— n matrices A, B, and
+ * C, a general problem is to verify whether AB=C
+ *
+ * The time Complexity of the algorithm is O(n^{2}).
+ */
+#include <array>     //for std::array
+#include <cassert>   //for assert
+#include <iostream>  //for io operations
+
+/**
+ * Freivald's algorithm
+ * The function will will check AB=C
+ * @param A matrix A
+ * @param B matrix B
+ * @param C resultant matrix C
+ * @returns boolean value
+ */
+template <std::size_t N>
+bool CoppersmithAlgo(std::array<std::array<double, N>, N> A,
+                     std::array<std::array<double, N>, N> B,
+                     std::array<std::array<double, N>, N> C) {
+    // Printing matrix A
+    std::cout << "\nThe first matrix is:\n ";
+    for (int i = 0; i < N; i++) {
+        for (int j = 0; j < N; j++) {
+            std::cout << A[i][j] << "\t";
+        }
+        std::cout << "\n";
+    }
+
+    // Printing matrix B
+    std::cout << "\nThe Second matrix is:\n ";
+    for (int i = 0; i < N; i++) {
+        for (int j = 0; j < N; j++) {
+            std::cout << B[i][j] << "\t";
+        }
+        std::cout << "\n";
+    }
+
+    // Printing matrix C
+    std::cout << "The resultant matrix: \n";
+    for (int i = 0; i < N; i++) {
+        for (int j = 0; j < N; j++) {
+            std::cout << C[i][j] << "\t";
+        }
+        std::cout << "\n";
+    }
+    // generate a random nx1 vector r containing 0/1 elements
+    std::array<std::array<const double, N>, 1> a{0};
+
+    // Multiply matrix B with vector 'r' which gives a nx1 vector 'br'
```

Comment:
instead of writing the multiplication logic again and again it will be better if you create a function
`multiply(mat A, mat B)`
to return the multiplication result, this way code looks more readable and clear and it also reduces the size of code